### ISSUE TITLE
Do not emit consider-iterating-dictionary in Python2.x when keys are assigned in loop body

### ISSUE THREAD

**Issue by rogalski** [PyCQA/pylint#1615]
### Steps to reproduce
```python
d = {'a': 1, 'b': 2, 'c': 3}
for k in d.keys():
    d[k+k] = 2*d[k]
```

### Current behavior
consider-iterating-dictionary warning

### Expected behavior
Message not emitted, without call to `.keys()` a RuntimeError is raised in linted code. Therefore, call to keys serves a purpose and shouldn't be treated as error.

### pylint --version output
master

**Comment by PCManticore** [comment]
I wouldn't worry about the Python 2 behaviour for now, since we're going to drop it hopefully as close as next year. But nonetheless, even for Python 3, the message doesn't make a lot of sense since we're getting RuntimeError with and without ``.keys()``.
